db.go

```diff
@@ -131,6 +136,34 @@ func (rr *redisRepo) fetchInstanceEvents(instanceID string) ([]*lifecycleEvent,
 	return events, nil
 }
 
+func (rr *redisRepo) fetchAllInstanceEvents() (map[string][]*lifecycleEvent, error) {
+	conn := rr.cg.Get()
+	defer rr.closeConn(conn)
+
+	instanceEventKeys, err := redis.Strings(conn.Do("KEYS", fmt.Sprintf("%s:instance:*:events", RedisNamespace)))
```

Comment:
I would agree if this were a hot path, but I expect this to be used sporadically by humans.  I was hoping to avoid having to add a separate set.